src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -187,6 +189,7 @@ abstract class ClassfileParser {
         }
       }
     }
+    private val in1: AbstractFileReader = new AbstractFileReader(java.util.Arrays.copyOfRange(in.buf, initBp, in.bp))
```

Comment:
Alternatively, we could expose `AbstractFileReader.getChar` and friends as static methods that accept the `byte[]` as an extra parameter. That would avoid the need to have a full `AbstractFileReader` here which is clunky.

Reply:
Maybe we can have a subtype of `AbstractFileReader` that takes an offset parameter, to avoid passing it everywhere explicitly (`in.x(y - initBp)`)

Reply:
And maybe mutate the array in the `in`  reader instead of having two readers around and setting one to null?

Reply:
These details aren't relevant after my later refactorings.